sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -661,11 +664,11 @@ Write withBigtableService(BigtableService bigtableService) {
     }
 
     @Override
-    public PDone expand(PCollection<KV<ByteString, Iterable<Mutation>>> input) {
+    public PCollection<BigtableWriteResult> expand(
```

Comment:
Let's add a "BigTableIO.write().withWriteResults()" method that return the updated transform so that this change does not break backwards compatibility. Also, please add documentation there describing this feature.

Reply:
alright, I'll pick this up later when I have more free time.  It'll be a bunch of refactoring to plumb this around like TextIO does, essentially copying all the options onto another transform and proxying them through.

Reply:
I think you don't necessarily need to copy all the options. You'd only need that if you want to support code like "BigTableIO.write().withWriteResults().withTable()..and so on", but if you're ok with allowing .withWriteResults() only in the last position on the builder, then you can just capture the whole original Write object, and the object returned by withWriteResults() doesn't need any more setters of its own. It's still a bit of refactoring, but not as much boilerplate.